fc/mef.py

```diff
@@ -217,265 +52,152 @@ def clustering_gmm(data, n_clusters = 8, initialization = 'distance_sub',
-            .format(initialization))
+            covars.append(np.cov(data_i.T))
+    means = np.array(means)
+
+    ###
+    # Run Gaussian Mixture Model Clustering
+    ###
+
+    # Initialize GMM object
+    gmm = GMM(n_components=n_clusters,
+              tol=tol,
+              min_covar=min_covar,
+              covariance_type='full',
+              params='mc',
+              init_params='')
+
+    # Set initial parameters
+    gmm.weight_ = weights
+    gmm.means_ = means
+    gmm.covars_ = covars
 
     # Fit 
     gmm.fit(data)
-    # Get labels using the responsibilities
+
+    # Get labels by sampling from the responsibilities
     # This avoids the complete elimination of a cluster if two or more clusters
     # have very similar means.
     resp = gmm.predict_proba(data)
-    labels = [np.random.choice(range(n_clusters), p = ri) for ri in resp]
+    labels = [np.random.choice(range(n_clusters), p=ri) for ri in resp]
 
     return labels
 
-def find_peaks_smoothed_mode(data, min_val = 0, max_val = 1023):
+def selection_std(populations,
+                  low=None,
+                  high=None,
+                  n_std_low=2.5,
+                  n_std_high=2.5):
     """
-    Find the mode of a dataset by finding the peak of a smoothed histogram.
+    Select populations if most of their elements are between two values.
 
-    This function finds the mode of a dataset by calculating the histogram,
-    using a 1D Gaussian filter to smooth out the histogram, and identifying
-    the maximum value of the smoothed histogram. The ``sigma`` parameter of
-    the Gaussian filter is taken from the standard deviation of `data`.
+    This function selects populations from `populations` if their means are
+    more than `n_std_low` standard deviations greater than `low`, and
+    `n_std_high` standard deviations lower than `high`.
 
     Parameters
     ----------
-    data : array
-        Nx1 array to calculate the mode from. `data` is assumed to contain
-        only integers.
-    min_val : int, optional
-        Minimum possible value in `data`.
-    max_val : int, optional
-        Maximum possible value in `data`.
+    populations : list of arrays or FCSData objects
+        Populations to select or discard.
+    low, high : int or float, optional
+        Left and right threshold. If None, use 0.015 of the lowest value
+        and 0.0985 of the highest value in the first population's `domain`,
+        if available.
+    n_std_low, n_std_high : float, optional
+        Number of standard deviations from `low` and `high`, respectively,
+        that a population's mean has to be closer than to be discarded.
 
     Returns
     -------
-    peak : float
-        Mode of `data`.
-    hist_smooth : array
-        ``(max_val - min_val + 1)``-long array containing the smoothed
-        histogram.
+    selected_mask : boolean array
+        Set of flags indicating whether a population has been selected.
```

Comment:
Hesitate to describe it as a set when we're not using a set. boolean array is clearly indicated above, so this isn't a real big issue, but I'd just say "Flags indicating ...", get rid of any reference to a data structure.